GameFramework/Algorithm/Algorithm.h

```diff
@@ -88,8 +88,8 @@ namespace gameframework
 		{
 			float progressRatio = static_cast<float>(val) / static_cast<float>(cycleVal);
 
-			//!cos波を調整し0~1を行き来する値を作成
-			return (cos(2.0f * D3DX_PI * progressRatio) + 1.0f) * 0.5f;
+			//sin波を調整し0~1を行き来する値を作成
+			return sin(D3DX_PI * progressRatio);
```

Comment:
それだと、

|progressRatio|2.0f * D3DX_PI * progressRatio|戻り値|
|---|---|---|
|0.0f|0.0f|0.0f|
|0.25f|1.0f|0.7f|
|0.5f|1.0f|1.0f|
|0.75f|0.7f|0.7f|
|1.0f|0.0f|0.0f|

になりませんか？

0.0f <= progressRatio <= 1.0f なのであれば
```suggestion
			progressRatio /= 2;
			return sin(D3DX_PI * progressRatio);
```
では？

Reply:
仕様として0から1ではなく
０～１～０が一周期としています

Reply:
なるほど。
であれば、これを呼んでいる `SwitchMinBetweenMax()` なりVerticesクラスなりを直さないと一旦maxまで行ってから小さくなってしまいますね。

Reply:
cosからsinに修正したので
1～0～1が0～1～0に戻ったのでminからmaxになっていると思います

Reply:
テスト追加したので動かしてみて下さい

Reply:
ビルドエラーなどで少し手間取っているので学校で教えてもらえるとありがたいです

Reply:
はい、ゲームの方に影響が無いレベルで動いているのなら後回しでも良いかも知れませんね。
こっちのリポジトリは、就活作品としての「まともなライブラリ」を作ろうという取り組みですので。(まぁ、いつかはやらなきゃだけど)